### PR TITLE
Improve logging

### PR DESCRIPTION
**Pull request by LeiQL**
- Improve logging in API/COA
- Increase go test timeout to 35 seconds as connection default 30 seconds timing out in the MQTT  `TestInitFailed` test and go test timeout should take a longer time.

### CODE REVIEW

File: api/pkg/apis/v1alpha1/providers/target/azure/adu/adu.go [hunk 1]
```diff
@@ -131,10 +132,10 @@ func (i *ADUTargetProvider) Get(ctx context.Context, dep model.DeploymentSpec, r
 	var err error = nil
 	defer observ_utils.CloseSpanWithError(span, &err)
 
-	sLog.Info("~~~ ADU Update Provider ~~~ : getting components")
+	sLog.Infof("P (ADU Target Provider): getting components: %s - %s, traceId: %s", dep.Instance.Scope, dep.Instance.Name, span.SpanContext().TraceID().String())
 	deployment, err := i.getDeployment()
 	if err != nil {
-		sLog.Errorf("~~~ ADU Target Provider ~~~ : %+v", err)
+		sLog.Errorf("P (ADU Target Provider): %+v, traceId: %s", err, span.SpanContext().TraceID().String())
```
Comment (Haishi2016) [inline api/pkg/apis/v1alpha1/providers/target/azure/adu/adu.go:138]: Add two spaces for indentation

File: api/pkg/apis/v1alpha1/providers/target/azure/iotedge/iotedge.go [hunk 2]
```diff
@@ -204,13 +204,13 @@ func (i *IoTEdgeTargetProvider) Get(ctx context.Context, deployment model.Deploy
 			var twin ModuleTwin
 			twin, err = i.getIoTEdgeModuleTwin(ctx, k)
 			if err != nil {
-				sLog.Error("  P(IoT Edge Target): +%v", err)
+				sLog.Errorf("  P(IoT Edge Target): failed to get %s module: +%v, traceId: %s", k, err, span.SpanContext().TraceID().String())
```
Comment (Haishi2016) [inline api/pkg/apis/v1alpha1/providers/target/azure/iotedge/iotedge.go:207]: Missing space between P and ( in this file

File: api/pkg/apis/v1alpha1/providers/target/http/http.go [hunk 3]
```diff
@@ -88,7 +88,7 @@ func (i *HttpTargetProvider) Get(ctx context.Context, deployment model.Deploymen
 	var err error = nil
 	defer observ_utils.CloseSpanWithError(span, &err)
 
-	sLog.Infof("  P(HTTP Target): getting artifacts: %s - %s", deployment.Instance.Scope, deployment.Instance.Name)
+	sLog.Infof("  P(HTTP Target): getting artifacts: %s - %s, traceId: %s", deployment.Instance.Scope, deployment.Instance.Name, span.SpanContext().TraceID().String())
```
Comment (Haishi2016) [inline api/pkg/apis/v1alpha1/providers/target/http/http.go:91]: Missing space between P and ( in this file

File: api/pkg/apis/v1alpha1/providers/target/ingress/ingress.go [hunk 4]
```diff
@@ -160,25 +160,25 @@ func (i *IngressTargetProvider) Init(config providers.IProviderConfig) error {
...
 	if err != nil {
-		sLog.Errorf("  P (Ingress Target): %+v", err)
+		sLog.Errorf("  P (Ingress Target): failed to create a dynamic client: %+v", err)
 		return err
 	}
 
 	i.DiscoveryClient, err = discovery.NewDiscoveryClientForConfig(kConfig)
 	if err != nil {
-		sLog.Errorf("  P (Ingress Target): %+v", err)
+		sLog.Errorf("  P (Ingress Target): failed to create a discovery client: %+v", err)
```
Comment (Haishi2016) [inline api/pkg/apis/v1alpha1/providers/target/ingress/ingress.go:181]: Should we add trace ids in this file as well?

File: api/pkg/apis/v1alpha1/providers/target/kubectl/kubectl.go [hunk 5]
```diff
@@ -125,7 +125,7 @@ func (i *KubectlTargetProvider) Init(config providers.IProviderConfig) error {
 
 	updateConfig, err := toKubectlTargetProviderConfig(config)
 	if err != nil {
-		sLog.Errorf("  P (Kubectl Target): expected KubectlTargetProviderConfig - %+v", err)
+		sLog.Errorf("  P (Kubectl Target): expected KubectlTargetProviderConfig - %+v")
```
Comment (Haishi2016) [inline api/pkg/apis/v1alpha1/providers/target/kubectl/kubectl.go:128]: Not supplying err object

File: api/pkg/apis/v1alpha1/providers/target/kubectl/kubectl.go [hunk 6]
```diff
@@ -165,25 +165,25 @@ func (i *KubectlTargetProvider) Init(config providers.IProviderConfig) error {
...
 	if err != nil {
-		sLog.Errorf("  P (Kubectl Target): %+v", err)
+		sLog.Errorf("  P (Kubectl Target): failed to create a dynamic client: %+v", err)
 		return err
 	}
 
 	i.DiscoveryClient, err = discovery.NewDiscoveryClientForConfig(kConfig)
 	if err != nil {
-		sLog.Errorf("  P (Kubectl Target): %+v", err)
+		sLog.Errorf("  P (Kubectl Target): failed to create a discovery client: %+v", err)
```
Comment (Haishi2016) [inline api/pkg/apis/v1alpha1/providers/target/kubectl/kubectl.go:186]: Trace ids?

File: api/pkg/apis/v1alpha1/providers/target/kubectl/kubectl.go [hunk 7]
```diff
@@ -323,28 +324,28 @@ func (i *KubectlTargetProvider) Apply(ctx context.Context, deployment model.Depl
 						case dataBytes, ok := <-chanMes:
 							if !ok {
 								err = errors.New("failed to receive from data channel")
-								sLog.Error("  P (Kubectl Target):  +%v", err)
+								sLog.Errorf("  P (Kubectl Target):  +%v, traceId: %s", err, span.SpanContext().TraceID().String())
```
Comment (Haishi2016) [inline api/pkg/apis/v1alpha1/providers/target/kubectl/kubectl.go:327]: I think the format string should be %+v, apply to the whole file

File: api/pkg/apis/v1alpha1/providers/target/win10/sideload/sideload.go [hunk 8]
```diff
@@ -94,10 +94,11 @@ func (i *Win10SideLoadProvider) Init(config providers.IProviderConfig) error {
 	var err error = nil
 	defer observ_utils.CloseSpanWithError(span, &err)
 
-	sLog.Info("~~~ Win 10 Sideload Provider ~~~ : Init()")
+	sLog.Info(" P (Win10Sideload Target): Init()")
```
Comment (Haishi2016) [inline api/pkg/apis/v1alpha1/providers/target/win10/sideload/sideload.go:97]: missing one space before P

File: api/pkg/apis/v1alpha1/providers/target/ingress/ingress.go (hunk 4 above)
Comment (LeiQL) [reply api/pkg/apis/v1alpha1/providers/target/ingress/ingress.go:181]: No trace ID in method `Init()`. `Context.TODO()` in the span initialization:
````
_, span := observability.StartSpan(
		"Ingress Target Provider",
		context.TODO(),
		&map[string]string{
			"method": "Init",
		},
	)
````
Trace included in other method in this file.

File: api/pkg/apis/v1alpha1/providers/target/kubectl/kubectl.go (hunk 6 above)
Comment (LeiQL) [reply api/pkg/apis/v1alpha1/providers/target/kubectl/kubectl.go:186]: No trace Id for `Init()` method as above mentioned

**Pull request merged**